### PR TITLE
ddtrace/tracer: add WithDebugStack StartOption to configure stack traces.

### PR DESCRIPTION
**Pull request by knusbaum**
By default a stack trace is collected when a span is finished with an
error. This can be disabled by passing tracer.NoDebugStack to
span.Finish(). Sometimes it is desirable to disable this option globally,
which is now possible with the WithDebugStack StartOption that this commit
introduces.

### CODE REVIEW

File: ddtrace/tracer/tracer_test.go [hunk 1]
```diff
@@ -59,7 +60,7 @@ loop:
 	}
 }
 
-// TestTracerFrenetic does frenetic testing in a scenario where the tracer is started
+// TestTracerCleanStop does frenetic testing in a scenario where the tracer is started
```
Comment (knusbaum) [inline ddtrace/tracer/tracer_test.go:63]: Unrelated doc change

File: ddtrace/tracer/span.go [hunk 2]
```diff
@@ -66,6 +66,7 @@ type span struct {
 	TraceID  uint64             `msg:"trace_id"`          // identifier of the root span
 	ParentID uint64             `msg:"parent_id"`         // identifier of the span's direct parent
 	Error    int32              `msg:"error"`             // error status of the span; 0 means no errors
+	errCfg   errorConfig        `msg:"-"`                 // configuration for errors tracing
```
Comment (gbbr) [inline ddtrace/tracer/span.go:69]: Should we just pass the whole config pointer? Does it make sense? Fine as is too.

**Review by gbbr — DISMISSED** [review]

File: ddtrace/tracer/span.go (hunk 2 above)
Comment (knusbaum) [reply ddtrace/tracer/span.go:69]: You mean having a `FinishConfig` rather than an `errorConfig`? 
That's a fine idea, and we can do more with options in the future if we want.

Comment (knusbaum) [reply ddtrace/tracer/span.go]: Hmm... I ended up removing both `FinishConfig` and `errorConfig`. We were only using one field from each, so it seemed counter-intuitive to include them. 

I still think it could be a good idea, but we can follow up in another issue/PR.

**Review by gbbr — APPROVED** [review]

File: ddtrace/tracer/span.go (hunk 2 above)
Comment (gbbr) [reply ddtrace/tracer/span.go]: Lgtm
